lib/Dialect/Random/IR/RandomTypes.td

```diff
@@ -0,0 +1,32 @@
+#ifndef HEIR_LIB_DIALECT_RANDOM_IR_RANDOMTYPES_TD_
+#define HEIR_LIB_DIALECT_RANDOM_IR_RANDOMTYPES_TD_
+
+include "RandomDialect.td"
+
+include "mlir/IR/DialectBase.td"
+include "mlir/IR/AttrTypeBase.td"
+
+// A base class for all types in this dialect
+class Random_Type<string name, string typeMnemonic>
+    : TypeDef<Random_Dialect, name> {
+  let mnemonic = typeMnemonic;
+}
+
+def Distribution : Random_Type<"Distribution", "distribution"> {
+  let summary = "A random distribution";
+
+  let description = [{
```

Comment:
since type conversion during lowering can't access the context, we probably should have an parameter allowing one of (uniform, discrete_gaussian) on the distribution type